src/App.tsx

```diff
@@ -1,61 +1,119 @@
 import './App.scss';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(userId: number) {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const newTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+let maxId = Math.max(...newTodos.map(todo => todo.id));
+
+export const App: React.FC = () => {
+  const [activeUser, setActiveUser] = useState(0);
+  const [message, setMessage] = useState('');
+  const [visibleTodos, setVisibleTodos] = useState(newTodos);
+  const [hasMessageError, setHasMessageError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const newTodo = () => {
+    maxId += 1;
+
+    return {
+      id: maxId,
+      title: message,
+      completed: false,
+      userId: activeUser,
+      user: getUser(activeUser),
+    };
+  };
+
+  const handleAdd = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setHasMessageError(!message);
+    setHasUserError(!activeUser);
+
+    if (message && activeUser) {
+      const updatedTodo = newTodo();
+
+      setVisibleTodos([
+        ...visibleTodos,
+        updatedTodo,
+      ]);
+    }
+
+    setMessage('');
+    setActiveUser(0);
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form action="/api/users" method="Get" onSubmit={handleAdd}>
```

Comment:
You don't need action and method, you are not sending or getting anything from the server, anyway you are preventing the default behavior.